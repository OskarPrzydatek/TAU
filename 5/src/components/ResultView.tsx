import { Operation } from '../consts/operation.enum';

type ResultViewProps = {
  alpha: number;
  beta: number;
  operation: Operation | null;
  result: number | null;
  resultTestID?: string;
  messageTestID?: string;
};

export default function ResultView({
  alpha,
  beta,
  operation,
  result,
  resultTestID,
  messageTestID,
}: ResultViewProps) {
  const isResultDifferrentFromNull = result !== null;

  return (
    <>
      {isResultDifferrentFromNull ? (
        <h2 data-testid={resultTestID}>
          {alpha} {operation} {beta} = {result}
        </h2>
      ) : (
        <h2 data-testid={messageTestID}>
          Put numbers to inputs and choose operation, then You'll see the result
          :D
        </h2>
      )}
    </>
  );
}
