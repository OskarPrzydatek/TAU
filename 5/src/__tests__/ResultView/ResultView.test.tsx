import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ResultView from '../../components/ResultView';
import { Operation } from '../../consts/operation.enum';

describe('ResultView', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <ResultView
          alpha={0}
          beta={0}
          operation={Operation.ADD}
          result={null}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure result is not shown when equals null', () => {
    const { rerender } = render(
      <ResultView
        alpha={0}
        beta={0}
        operation={Operation.ADD}
        result={null}
        resultTestID="result"
        messageTestID="message"
      />
    );
    expect(screen.queryByTestId('result')).toBeNull();
    expect(screen.queryByTestId('message')).not.toBeNull();
    rerender(
      <ResultView
        alpha={3}
        beta={3}
        operation={Operation.ADD}
        result={6}
        resultTestID="result"
        messageTestID="message"
      />
    );
    expect(screen.queryByTestId('message')).toBeNull();
    expect(screen.queryByTestId('result')).not.toBeNull();
    expect(screen.getByTestId('result')).toHaveTextContent('3 + 3 = 6');
  });
});
