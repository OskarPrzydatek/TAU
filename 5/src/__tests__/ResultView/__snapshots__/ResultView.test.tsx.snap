// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultView Component Snapshot 1`] = `
<h2>
  Put numbers to inputs and choose operation, then You'll see the result :D
</h2>
`;
